lib/omniauth/strategies/saml.rb

```diff
@@ -103,15 +109,23 @@ def other_phase
       uid { @name_id }
 
       info do
-        {
-          :name  => @attributes[:name],
-          :email => @attributes[:email] || @attributes[:mail],
-          :first_name => @attributes[:first_name] || @attributes[:firstname] || @attributes[:firstName],
-          :last_name => @attributes[:last_name] || @attributes[:lastname] || @attributes[:lastName]
-        }
+        found_attributes = options.attribute_statements.map do |key, values|
+          attribute = find_attribute_by(values)
+          [key, attribute]
```

Comment:
I'd still like to see this scheme extended to support multi-valued attributes as suggested in #36, so perhaps a similar approach could be used to add a `:required => true` flag.